File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/CheckReorder/MainStreetCheckReorder.cs
namespace Psi.Data.Models.ClientConfigurationModels.CheckReorder
{
    public class MainStreetCheckReorder : SettingsBaseHelper
    {
        public MainStreetCheckReorder(ISettingsBase settingsBase) : base(settingsBase)
        {
        }

        [SettingKey("CheckReorder.MainStreetCheckReorder.Enabled")]
        public bool Enabled
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        [SettingKey("CheckReorder.MainStreetCheckReorder.MinVersion")]
        public double MinVersion
        {
            get => GetDoubleValue();
            set => SetValue(value);
        }

        [SettingKey("CheckReorder.MainStreetCheckReorder.SsoUrl")]
        public string SsoUrl
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("CheckReorder.MainStreetCheckReorder.BankId")]
        public string BankId
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("CheckReorder.MainStreetCheckReorder.BranchNumber")]
        public string BranchNumber
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("CheckReorder.MainStreetCheckReorder.UserId")]
        public string UserId
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("CheckReorder.MainStreetCheckReorder.Key")]
        public string Key
        {
            get => GetValue();
            set => SetValue(value);
        }
    }
}

File: workspace/legacy-analyzer/output/infrastructure/models/Psi/Data/Models/ClientConfigurationModels/CheckReorder/MainStreetCheckReorder.ts
import { Setting, ISettingsGroup, ISettingsMetadata } from '@models/base/types';
export interface MainStreetCheckReorderConfig {
    Enabled: boolean;
    MinVersion: number;
    SsoUrl: string;
    BankId: string;
    BranchNumber: string;
    UserId: string;
    Key: string;
}

export class MainStreetCheckReorder implements ISettingsGroup {
    static readonly metadata: ISettingsMetadata = {
        groupName: 'MainStreetCheckReorder'
    };


            private _enabled: boolean;
            get enabled(): boolean {
                return this._enabled;
            }
            set enabled(value: boolean) {
                this._enabled = value;
            }

            private _minVersion: number;
            get minVersion(): number {
                return this._minVersion;
            }
            set minVersion(value: number) {
                this._minVersion = value;
            }

            private _ssoUrl: string;
            get ssoUrl(): string {
                return this._ssoUrl;
            }
            set ssoUrl(value: string) {
                this._ssoUrl = value;
            }

            private _bankId: string;
            get bankId(): string {
                return this._bankId;
            }
            set bankId(value: string) {
                this._bankId = value;
            }

            private _branchNumber: string;
            get branchNumber(): string {
                return this._branchNumber;
            }
            set branchNumber(value: string) {
                this._branchNumber = value;
            }

            private _userId: string;
            get userId(): string {
                return this._userId;
            }
            set userId(value: string) {
                this._userId = value;
            }

            private _key: string;
            get key(): string {
                return this._key;
            }
            set key(value: string) {
                this._key = value;
            }


    constructor() {}


        toSettings(): Setting[] {
            return [
                { key: "MainStreetCheckReorder.Enabled", value: this._enabled, dataType: 'boolean', label: "Enabled" },
                { key: "MainStreetCheckReorder.MinVersion", value: this._minVersion, dataType: 'number', label: "Min Version" },
                { key: "MainStreetCheckReorder.SsoUrl", value: this._ssoUrl, dataType: 'string', label: "Sso Url" },
                { key: "MainStreetCheckReorder.BankId", value: this._bankId, dataType: 'string', label: "Bank Id" },
                { key: "MainStreetCheckReorder.BranchNumber", value: this._branchNumber, dataType: 'string', label: "Branch Number" },
                { key: "MainStreetCheckReorder.UserId", value: this._userId, dataType: 'string', label: "User Id" },
                { key: "MainStreetCheckReorder.Key", value: this._key, dataType: 'string', label: "Key" },
            ];
        }

}